fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/socialmedia/SocialMediaViewModel.kt

```diff
@@ -225,36 +227,50 @@ constructor(
-              for (id in ids) {
-                chirps.add(socialMediaRepository.getChirp(laoId, id))
-              }
-              chirps
-            }
-            // Zip the subjects together to a sorted list
-            .flatMap { observables: List<Observable<Chirp>> ->
-              Observable.combineLatest(observables) { chirps: Array<Any?> ->
-                Arrays.stream(chirps)
-                    .map { obj: Any? -> Chirp::class.java.cast(obj) }
-                    .sorted(
-                        Comparator.comparing { chirp: Chirp? ->
-                          if (chirp != null) -chirp.timestamp else 0
-                        })
-                    .collect(Collectors.toList())
-              }
-            }
-            // We want to observe these changes on the main thread such that any modification done
-            // to
-            // the view are done on the thread. Otherwise, the app might crash
-            .observeOn(schedulerProvider.mainThread())
+    get() {
+      val laoIds = listOf(laoId) + linkedOrgRepo.getLinkedLaos(laoId).keys
+      val laoObservables: MutableList<Observable<MutableList<Chirp>>> = ArrayList()
+      for (lao in laoIds) {
+        val observable =
+            socialMediaRepository
+                .getChirpsOfLao(lao)
+                .map { ids: Set<MessageID> ->
+                  val chirps: MutableList<Observable<Chirp>> = ArrayList(ids.size)
+                  for (id in ids) {
+                    chirps.add(socialMediaRepository.getChirp(lao, id))
+                  }
+                  chirps
+                }
+                .flatMap { observables: List<Observable<Chirp>> ->
+                  Observable.combineLatest(observables) { chirps: Array<Any?> ->
+                    Arrays.stream(chirps)
+                        .map { obj: Any? -> Chirp::class.java.cast(obj) }
+                        .sorted(
+                            Comparator.comparing { chirp: Chirp? ->
+                              if (chirp != null) -chirp.timestamp else 0
+                            })
+                        .collect(Collectors.toList())
+                  }
+                }
+        laoObservables.add(observable)
+      }
+      val combinedObservables =
+          Observable.combineLatest(laoObservables) { chirpsLists: Array<Any?> ->
+            Arrays.stream(chirpsLists)
+                .flatMap { chirpsList: Any? -> (chirpsList as List<Chirp>).stream() }
+                .sorted(
+                    Comparator.comparing { chirp: Chirp? ->
+                      if (chirp != null) -chirp.timestamp else 0
+                    })
+                .collect(Collectors.toList())
+          }
+      // We want to observe these changes on the main thread such that any modification done
+      // to the view are done on the thread. Otherwise, the app might crash
+      return combinedObservables.observeOn(schedulerProvider.mainThread())
+    }
```

Comment:
This getter will be executed every time we access that field, which looks very heavy as getter method. You could precompute this variable in initialization, such that this function is only called once

Reply:
Yeah sorry I was missing the context of execution of that function, now I looked it up better and it's how it's supposed to work, even if it's not very efficient and perhaps in the future we could change it